src/main/java/ch/njol/skript/expressions/ExprStringColor.java

```diff
@@ -0,0 +1,206 @@
+package ch.njol.skript.expressions;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.expressions.base.PropertyExpression;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.lang.SyntaxStringBuilder;
+import ch.njol.skript.util.ColorRGB;
+import ch.njol.skript.util.SkriptColor;
+import ch.njol.util.Kleenean;
+import ch.njol.util.coll.CollectionUtils;
+import org.bukkit.event.Event;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Name("String Colors")
+@Description({
+	"Retrieve the first, the last, or all of the colors of a string.",
+	"NOTE: The retrieved colors of the string will be formatted with the color symbol."
+})
+@Examples({
+	"set {_colors::*} to the string colors of \"<red>hey<blue>yo\"",
+	"",
+	"set {_color} to the first string color code of \"&aGoodbye!\"",
+	"send \"%{_color}%Howdy!\" to all players"
+})
+@Since("INSERT VERSION")
+public class ExprStringColor extends PropertyExpression<String, Object> {
+
+	private enum StringColor {ALL, FIRST, LAST;}
+
+	private static final StringColor[] STRING_COLORS = StringColor.values();
+
+	static {
+		Skript.registerExpression(ExprStringColor.class, Object.class, ExpressionType.PROPERTY,
+			"[all [of the|the]|the] string color[s] [code:code[s]] of %strings%",
+			"[the] first string color[s] [code:code[s]] of %strings%",
+			"[the] last string color[s] [code:code[s]] of %strings%");
+	}
+
+	private StringColor selectedState;
+	private boolean getCodes;
+
+	@Override
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		selectedState = STRING_COLORS[matchedPattern];
+		getCodes = parseResult.hasTag("code");
+		//noinspection unchecked
+		setExpr((Expression<String>) exprs[0]);
+		return true;
+	}
+
+	@Override
+	protected Object @Nullable [] get(Event event, String[] source) {
+		List<Object> colors = new ArrayList<>();
+		for (String string : getExpr().getArray(event)) {
+			List<Object> stringColors = getColors(string);
+			if (stringColors.isEmpty())
+				continue;
+			colors.addAll(stringColors);
+		}
+		return colors.toArray(new Object[0]);
+	}
+
+	@Override
+	public Class<Object> getReturnType() {
+		return Object.class;
```

Comment:
you can return the actual class that will be returned, since it depends on the tag in init.